### ISSUE TITLE
Display release number and environment in page footer

### ISSUE THREAD

**Issue by lamontfr** [savoirfairelinux/sous-chef#743]
### Expected Behaviour
- As a Customer manager at Santropol Roulant, I should be able to know the release number of Souschef currently running.
- As a Staff member at Santropol Roulant, I should be able to see easily in the UI if I am connected to the Production environment or to the Test environment.

### Actual Behaviour
- I have to ask the sysadmin about which release was installed on the server.

### Suggestions to solve issue
- In the base template page footer, display the release number which is kept in a constant in a configuration file that is updated at each release.
- The release number should contain a prefix identifying the environment (Prod or Test).

**Comment by kousu** [comment]
I think that the git revision (`git show-ref HEAD | cut -f 1 -d " "`) should be embedded in the output, not the environment name which has to be manually set. Better, if the git revision also happens to be tagged, that should be shown instead, but I am not sure how to do that; I know people do that, because then you automatically know if you're at v1.0-beta2, v1.0-beta3, or somewhere in between (and which somewhere).

**Comment by kousu** [comment]
Thanks for solving this @lingxiaoyang! I am impressed that you found the more corre `git rev-parse HEAD` and `git describe --tags`. I never knew how to do that. Now I do :). And thanks @erozqba for doing the install this morning without me, that's excellent.

I have a puzzle though. This feature does not seem to be working on the dev server, and I can't tell why.

The commit is on the dev server:
```
sfl@sous-chef-test:~/sous-chef$ git log -p --decorate 4324350b039f2762d85be2c8f9442f5afef50802
commit 4324350b039f2762d85be2c8f9442f5afef50802
Author: Ernesto Rodriguez Ortiz <erozqba@gmail.com>
Date:   Tue Aug 15 22:31:52 2017 -0400

    Fixed #743: display git information. Add the work in #780 (#800)
    
    * Fixed #743: display git information.
    
    * Put in the footer the GIT HEAD and the GIT TAG

diff --git a/src/sous_chef/context_processors.py b/src/sous_chef/context_processors.py
index fa9ca7c..3096c25 100644
--- a/src/sous_chef/context_processors.py
+++ b/src/sous_chef/context_processors.py
@@ -30,6 +30,8 @@ def total(request):
         'NOTE_FILTER_DEFAULT_IS_READ': NoteFilter.NOTE_STATUS_UNREAD,
         'SC_ENVIRONMENT_NAME': settings.SOUSCHEF_ENVIRONMENT_NAME,
         'SC_VERSION': settings.SOUSCHEF_VERSION,
+        'GIT_HEAD': settings.GIT_HEAD,
+        'GIT_TAG': settings.GIT_TAG,
     }
 
     return COMMON_CONTEXT
diff --git a/src/sous_chef/settings.py b/src/sous_chef/settings.py
index 7202d18..fa00b7a 100644
--- a/src/sous_chef/settings.py
+++ b/src/sous_chef/settings.py
@@ -12,6 +12,7 @@ https://docs.djangoproject.com/en/1.9/ref/settings/
 """
 
 import os
+import subprocess
 from django.urls import reverse_lazy
 
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
@@ -222,3 +223,10 @@ AVATAR_PROVIDERS = (
 # Displayable information
 SOUSCHEF_VERSION = os.environ.get('SOUSCHEF_VERSION') or ''
 SOUSCHEF_ENVIRONMENT_NAME = os.environ.get('SOUSCHEF_ENVIRONMENT_NAME') or ''
+
+try:
+    GIT_HEAD = subprocess.check_output(['git', 'rev-parse', 'HEAD'])
+    GIT_TAG = subprocess.check_output(['git', 'describe', '--tags'])
+except Exception as e:
+    GIT_HEAD = None
+    GIT_TAG = None
...
```

Docker seems to have picked up the new code:
```
sfl@sous-chef-test:~/sous-chef$ docker-compose exec web grep GIT src/sous_chef/settings.py
WARNING: Some networks were defined but are not used by any service: frontend
    GIT_HEAD = subprocess.check_output(['git', 'rev-parse', 'HEAD'])
    GIT_TAG = subprocess.check_output(['git', 'describe', '--tags'])
    GIT_HEAD = None
    GIT_TAG = None
sfl@sous-chef-test:~/sous-chef$ 
```

And the git commands run fine inside of docker---even inside of python inside of docker:

```
sfl@sous-chef-test:~/sous-chef$ docker-compose exec web git rev-parse HEAD
WARNING: Some networks were defined but are not used by any service: frontend
6c27f2afeeb16b1916d6179a2d956e61170e1560
sfl@sous-chef-test:~/sous-chef$ docker-compose exec web python -c "import subprocess; print(subprocess.check_output(['git', 'rev-parse', 'HEAD']))"
WARNING: Some networks were defined but are not used by any service: frontend
b'6c27f2afeeb16b1916d6179a2d956e61170e1560\n'
sfl@sous-chef-test:~/sous-chef$ docker-compose exec web python -c "import subprocess; print(subprocess.check_output(['git', 'describe', '--tags']))"
WARNING: Some networks were defined but are not used by any service: frontend
b'v1.0-beta4-9-g6c27f2a\n'
sfl@sous-chef-test:~/sous-chef$ docker-compose exec web git describe --tags
WARNING: Some networks were defined but are not used by any service: frontend
v1.0-beta4-9-g6c27f2a
sfl@sous-chef-test:~/sous-chef$ 
```
But the version is not showing up in th UI:

![screenshot from 2017-08-16 16-54-44](https://user-images.githubusercontent.com/987487/29384974-0814b564-82a4-11e7-93f6-75644626061f.png)

I don't get it. The code looks _perfect_ to me.

**Comment by erozqba** [comment]
Hi Nick,

You are not looking into the right place, the git information is in the
footer of the page.

[image: Imágenes integradas 1]


Eroz

2017-08-16 16:58 GMT-04:00 Nick <notifications@github.com>:
